<launch>
  <arg name="svo_file"/>
  <arg name="zed_id"/>

  <include file="$(find autobot)/launch/zed_tf.launch" />
  <node name="autobot_zed" pkg="autobot" type="autobot_zed" output="screen" >

    <!-- SVO file path -->
    <!-- param name="svo_filepath" value="$(arg svo_file)" /-->

    <!-- ZED parameters -->
    <!--param name="zed_id"                value="$(arg zed_id)" /-->

    <param name="resolution"            value="3" />
    <param name="quality"               value="1" />
    <param name="sensing_mode"          value="1" />
    <param name="frame_rate"            value="30" />
    <param name="odometry_db"           value="" />
    <param name="openni_depth_mode"     value="0" />
    <param name="gpu_id"                value="-1" />

    <!-- ROS topic names -->
    <param name="rgb_topic"             value="rgb/image_rect_color" />
    <param name="rgb_raw_topic"         value="rgb/image_raw_color" />
    <param name="rgb_cam_info_topic"    value="rgb/camera_info" />

    <param name="left_topic"            value="left/image_rect_color" />
    <param name="left_raw_topic"        value="left/image_raw_color" />
    <param name="left_cam_info_topic"   value="left/camera_info" />

    <param name="right_topic"           value="right/image_rect_color" />
    <param name="right_raw_topic"       value="right/image_raw_color" />
    <param name="right_cam_info_topic"  value="right/camera_info" />

    <param name="depth_topic"           value="depth/depth_registered" />
    <param name="depth_cam_info_topic"  value="depth/camera_info" />

    <!--param name="point_cloud_topic"     value="point_cloud/cloud_registered"  /-->

    <!--param name="odometry_topic"        value="odom" /-->

  </node>
    <node name="obj_detector" pkg="autobot" type="obj_detector" args="--model /media/ubuntu/SSD/jetson-inference/data/networks/facenet-120/snapshot_iter_24000.caffemodel --prototxt /media/ubuntu/SSD/jetson-inference/data/networks/facenet-120/deploy.prototxt"/>

  <!-- ROS URDF description of the ZED -->
  <param name="zed_description" textfile="$(find autobot)/urdf/zed.urdf" />
  <node name="zed_state_publisher" pkg="robot_state_publisher" type="state_publisher">
    <remap from="robot_description" to="zed_description" />
  </node>

</launch>
